### PR TITLE
busco: compatibility with python 3.7

### PR DESCRIPTION
**Pull request by suujia**
- [ ] Have you followed the [guidelines for contributing](https://github.com/brewsci/homebrew-bio/blob/master/CONTRIBUTING.md)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/brewsci/homebrew-bio/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source FORMULA`, where `FORMULA` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict FORMULA` (after doing `brew install FORMULA`)?

-----

### CODE REVIEW

File: Formula/busco.rb [hunk 1]
```diff
@@ -19,8 +19,11 @@ class Busco < Formula
   # Also depends on augustus and hmmer
 
   def install
+    xy = Language::Python.major_minor_version "python3"
+    ENV.prepend_create_path "PYTHONPATH", libexec/"lib/python${xy}/site-packages"
+
     inreplace Dir["scripts/*.py"], "#!/usr/bin/env python", "#!#{HOMEBREW_PREFIX}/bin/python3"
-    system "python3", "setup.py", "install", "--prefix=#{prefix}"
+    system "python", "setup.py", "install", "--user", "--prefix="
```
Comment (sjackman) [inline Formula/busco.rb]: Leave this line as `python3` and change…
```diff
-depends_on "python3"
+depends_on "python"
```

Comment (sjackman) [reply Formula/busco.rb]: I believe `--prefix=…` and `--user` are mutually exclusive.

**Comment by suujia** [comment]
https://gitlab.com/ezlab/busco/issues/27
I made the following changes: 

1) 
```
I have never faced the problem myself, but what about 
python setup.py install --user --prefix= as I suggested. (yes empty prefix)
```
2) 
```
Another possible solution is to append the path to your globally installed BUSCO modules to PYTHONPATH.
```
When I looked in orca image location for site-packages, I found it here and added it to path: 
`export PYTHONPATH="${PYTHONPATH}:/.linuxbrew/Cellar/busco/3.0.2/lib/python3.6/site-packages"`
Even though the python that was downloaded in orca is python 3.7, so I think BUSCO couldn't find pipebricks.

File: Formula/busco.rb [hunk 2]
```diff
@@ -14,13 +14,17 @@ class Busco < Formula
...
   # Also depends on augustus and hmmer
 
   def install
+    xy = Language::Python.major_minor_version "python3"
+    ENV.prepend_create_path "PYTHONPATH", libexec/"lib/python${xy}/site-packages"
+    system python, "import pipebricks"
+
     inreplace Dir["scripts/*.py"], "#!/usr/bin/env python", "#!#{HOMEBREW_PREFIX}/bin/python3"
-    system "python3", "setup.py", "install", "--prefix=#{prefix}"
+    system "python3", "setup.py", "install", "--user", "--prefix="
```
Comment (sjackman) [inline Formula/busco.rb]: I believe `--user` will install somewhere in the user's home directory. We want to use `--prefix=#{prefix}` to install in a directory that we specify.

**Comment by suujia** [comment]
The changes worked! 😺

File: Formula/busco.rb [hunk 3]
```diff
@@ -14,11 +14,14 @@ class Busco < Formula
...
-  depends_on "python3"
+  depends_on "python"
   depends_on "blast" => :recommended
   # Also depends on augustus and hmmer
 
   def install
+    xy = Language::Python.major_minor_version "python3"
+    ENV.prepend_create_path "PYTHONPATH", libexec/"lib/python${xy}/site-packages"
+
     inreplace Dir["scripts/*.py"], "#!/usr/bin/env python", "#!#{HOMEBREW_PREFIX}/bin/python3"
```
Comment (sjackman) [inline Formula/busco.rb]: Is this changed needed, and does it work without it?

**Comment by sjackman** [comment]
Woohoo!

**Comment by suujia** [comment]
Yup, removing the PYTHONPATH command words as well! Thought that was what's breaking it but ok! :)

**Comment by sjackman** [comment]
Please add `revision 1` after `sha256`.

**Comment by suujia** [comment]
this should be good to go :)

**Comment by sjackman** [comment]
Merged. Thanks again, Susie! Have a good weekend! 🌞

**Pull request closed without merging**